x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteForecastAction.java

```diff
@@ -185,9 +186,11 @@ private void deleteForecasts(SearchResponse searchResponse,
         SearchHits hits = searchResponse.getHits();
         List<ForecastRequestStats> allStats = new ArrayList<>(hits.getHits().length);
         for (SearchHit hit : hits) {
-            XContentParser parser = XContentFactory.xContent(XContentType.JSON).createParser(
-                NamedXContentRegistry.EMPTY, DeprecationHandler.THROW_UNSUPPORTED_OPERATION, hit.getSourceRef().streamInput());
-            allStats.add(ForecastRequestStats.STRICT_PARSER.apply(parser, null));
+            try (InputStream stream = hit.getSourceRef().streamInput();
```

Comment:
Here, we don't need the stream in a separate variable. The parser owns it and will handle closing the stream.